gwcs/examples.py

```diff
@@ -0,0 +1,479 @@
+    n2c = models.RotateNative2Celestial(*crval, 180, name='sky_rotation')
+    cel_model = wcslin | tan | n2c
+    icrs = cf.CelestialFrame(reference_frame=coord.ICRS(), name='sky',
+                             axes_order=cel_axes_order)
+
+    wcs_forward = (cel_model & time_model) | models.Mapping(axes_order)
+
+    comp_frm = cf.CompositeFrame(frames=[icrs, time_frame],
+                                 name='TEST 3D FRAME WITH TIME')
+
+    w = wcs.WCS(forward_transform=wcs_forward, output_frame=comp_frm,
+                input_frame=detector_frame)
+
+    w.bounding_box = tuple((0, k - 1) for k in cube_size)
+    w.pixel_shape = cube_size
+    w.array_shape = w.pixel_shape[::-1]
+    return w
+
+
+def gwcs_7d_complex_mapping():
+    """
+    Useful features of this WCS (axes indices here are 0-based):
+        - includes two celestial axes: input (0, 1) maps to world (2 - RA, 1 - Dec)
+        - includes one separable frame with one axis: 4 -> 2
+        - includes one frame with 3 input and 4 output axes (1 degenerate),
+          with separable world axes (3, 5) and (0, 6).
+    """
+    offx = models.Shift(-64, name='x_translation')
+    offy = models.Shift(-32, name='y_translation')
+    cd = np.array([[1.2906, 0.59532], [0.50222, -1.2645]])
+    aff = models.AffineTransformation2D(matrix=1e-5 * cd, name='rotation')
+    aff2 = models.AffineTransformation2D(matrix=cd, name='rotation2')
+
+    wcslin = (offx & offy) | aff
+    tan = models.Pix2Sky_TAN(name='tangent_projection')
+    n2c = models.RotateNative2Celestial(5.630568, -72.0546, 180, name='skyrot')
+    icrs = cf.CelestialFrame(reference_frame=coord.ICRS(), name='sky', axes_order=(2, 1))
+    spec = cf.SpectralFrame(name='wave', unit=[u.m], axes_order=(4,), axes_names=('lambda',))
+    cmplx = cf.CoordinateFrame(
+        name="complex",
+        naxes=4,
+        axes_order=(3, 5, 0, 6),
+        axis_physical_types=(['em.wl', 'em.wl', 'time', 'time']),
+        axes_type=("SPATIAL", "SPATIAL", "TIME", "TIME"),
+        axes_names=("x", "y", "t", 'tau'),
+        unit=(u.m, u.m, u.second, u.second)
+    )
+
+    comp_frm = cf.CompositeFrame(frames=[icrs, spec, cmplx], name='TEST 7D')
+    wcs_forward = ((wcslin & models.Shift(-3.14) & models.Scale(2.7) & aff2) |
+                   (tan & models.Identity(1) & models.Identity(1) & models.Identity(2)) |
+                   (n2c & models.Identity(1) & models.Identity(1) & models.Identity(2)) |
+                   models.Mapping((3, 1, 0, 4, 2, 5, 3)))
+
+
+    detector_frame = cf.CoordinateFrame(
+        name="detector", naxes=6,
+        axes_order=(0, 1, 2, 3, 4, 5),
+        axes_type=("pixel", "pixel", "pixel", "pixel", "pixel", "pixel"),
+        unit=(u.pix, u.pix, u.pix, u.pix, u.pix, u.pix)
+    )
+
+    # pipeline = [('detector', wcs_forward), (comp_frm, None)]
+    w = wcs.WCS(forward_transform=wcs_forward, output_frame=comp_frm,
+                input_frame=detector_frame)
```

Comment:
Should this be updated to
```suggestion
    pipeline = [(detector_frame, wcs_forward), (comp_frm, None)]
    w = wcs.WCS(pipeline)
```

Reply:
I think the above is still an allowed way to initialize a WCS object. Let's keep it as is.